### ISSUE TITLE
Global styles doesn't apply on blocks that not belong to Gutenberg (e.g: WC Blocks)

### ISSUE THREAD

**Issue by gigitux** [WordPress/gutenberg#40808]
### Description

With the new Gutenberg versions (from 13.0) the Global Styles for WC Blocks plugin doesn’t apply anymore. It seems that the PR that introduce this bug is [#39559](https://github.com/WordPress/gutenberg/pull/39559/files): in particular, [this change](https://github.com/WordPress/gutenberg/pull/39559/files#diff-a68821415230540a4e41b902b125c3cc8ed44851c6f5a84817d37160076c51aaR148).

### Step-by-step reproduction instructions

1. Install WooCommerce.
2. Open the FSE Editor.
3. Add a WC Blocks (e.g: `Feature Product Block`).
4. Customize the global styles related to the block.
5. Save.
6. Notice that the Global Styles is not saved.

### Screenshots, screen recording, code snippet

![Screen Capture on 2022-05-04 at 10-17-39](https://user-images.githubusercontent.com/4463174/166645293-31153d5e-3a7c-431d-b73d-a839abf642cb.gif)

### Environment info

- WordPress 5.9, Gutenberg trunk version, Twenty Twenty-Two

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Aljullu** [comment]
@gziolo did you have time to look into this? We might spend some time next week investigating this with more detail, but wondering if some Gutenberg devs could give it a try as well to help diagnose if the issue is in our end or in Gutenberg.

**Comment by gziolo** [comment]
@Aljullu, I didn't have a chance to look into it. Is it also an issue with WordPress 6.0 RC4 without the Gutenberg plugin? There is different handling for webfonts being shipped in WordPress core so it would be great to clarify what the actual issue is happening.

> It seems that the PR that introduce this bug is [#39559](https://github.com/WordPress/gutenberg/pull/39559/files): in particular, [this change](https://github.com/WordPress/gutenberg/pull/39559/files#diff-a68821415230540a4e41b902b125c3cc8ed44851c6f5a84817d37160076c51aaR148).

Making sure I understood it correctly. The issue applies only to webfonts?

**Comment by gigitux** [comment]
> @Aljullu, I didn't have a chance to look into it. Is it also an issue with WordPress 6.0 RC4 without the Gutenberg plugin? There is different handling for webfonts being shipped in WordPress core so it would be great to clarify what the actual issue is happening.

No, I'm not able to replicate it with WordPress 6.0 RC4 without the Gutenberg plugin. With WordPress 6.0 RC4 **and** Gutenberg plugin, I'm able to replicate it. 

> Making sure I understood it correctly. The issue applies only to webfonts?

No, the issue is for all the Global Styles.

**Comment by oandregal** [comment]
Hi @gigitux I've reviewed this using the featured product block in WordPress 5.9 and 6.0 and I wasn't able to make it work in any version (tested frontend, post editor, and site editor: none worked). Plugins: latest WooCommerce but without Gutenberg.

This is what I've found:

- the global styles are correctly generated for all blocks in both versions, including the featured product block. The class generated for that block is `wp-block-woocommerce-featured-product`.
- Though the block uses the `wc-block-featured-product` class instead.

So the issue is that the class generated by global styles is not the one the block uses. The way to fix it is by reconciliating them. One way is to update the class in use by the block. The other way is to tell Global Styles to use a different class than the default it generates, [this is how it can be done](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#block-styles): 

- the block needs to have a block.json
- use the `__experimentalSelector` key with the selector it wants (`wc-block-featured-product` in this case)
- be registered in the server so this information is readily available for the styles engine to use

I'm going to close this issue as it's part of block configuration.

**Comment by danielwrobert** [comment]
Hi @oandregal! :wave:

I've dug into this a bit further and concluded that the class names for these blocks was not the root of the issue.

> Though the block uses the `wc-block-featured-product` class instead.

These blocks actually use both the `wc-block-featured-product` and the `wp-block-woocommerce-featured-product`.

You can read more on my findings [in this comment](https://github.com/woocommerce/woocommerce-blocks/pull/7000#:~:text=On%20this%20branch%2C%20with%20the%20latest%20version%20of%20Gutenberg%20active%2C%20the%20Global%20Styles%20are%20working%20in%20the%20admin%20but%20not%20rendering%20to%20the%20frontend.%20This%20is%20also%20the%20same%20behavior%20on%20our%20other%20blocks%2C%20as%20reported%20here%20(see%20screenshots%20below).) and in [this comment](https://github.com/woocommerce/woocommerce-blocks/issues/6988#issuecomment-1228547587) as well.

**TL;DR:**

- There was a separate issue where `__experimentalDuotone` was preventing the Global Styles from being applied properly. We've since removed that experimental property in [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/7000), for the time being.
- The original issue reported here still exists, however. If the latest Gutenberg plugin is active, the Global Styles for a non-Gutenberg block are not properly applied on the front-end. This is the case for _all_ of the Woo blocks and we have also been able to replicate the issue with the Jetpack blocks (see screenshots below).

| Plugins | Editor (Beta) | Frontend |
| ------ | ----- | ----- |
|  ![Screen Shot 2022-09-02 at 12 17 20 PM](https://user-images.githubusercontent.com/481776/188197022-2a80be44-9897-43aa-9b87-f8bd47c3b1ab.png)  |  ![Screen Shot 2022-09-02 at 12 18 15 PM](https://user-images.githubusercontent.com/481776/188197141-bb91cf9c-63e0-413a-bde6-00dab1cdf229.png)  |  ![Screen Shot 2022-09-02 at 12 18 31 PM](https://user-images.githubusercontent.com/481776/188197223-aeed094b-324f-47d1-8c49-2a9001da2e05.png) |

That being said, can we reopen this issue or would it be preferred that I create a new issue and crosslink to this one?

Thanks in advance!

cc @Aljullu, @gigitux

**Comment by oandregal** [comment]
Thanks for sharing your findings @danielwrobert

I've reopened. In [my testing](https://github.com/WordPress/gutenberg/issues/40808#issuecomment-1225975167), the styles for the block didn't work in 5.9 or 6.0 (no Gutenberg installed).

@ajlende @scruffian would you have some time to take a look at this?

**Comment by danielwrobert** [comment]
Thanks, @oandregal, I appreciate it!

There seems to have been a couple of separate-but-related issues at play (in addition to the aforementioned issue with `__experimentalDuotone`):

One issue was caused by the default styles for the text color on the Featured Items blocks preventing the Global Styles from taking effect. I’m working on [addressing that issue in this PR](https://github.com/woocommerce/woocommerce-blocks/pull/7036). It’s possible that this is what you running into with your testing?

The other issue is the problem reported here with the Global Styles not applying to non-Core blocks when the Gutenberg feature plugin is enabled - which seems impact all Woo blocks, along with Jetpack blocks.

**Comment by scruffian** [comment]
I'm looking at this, but I'm not 100% sure what you're expecting to see? Should third party blocks appear in the Global Styles interface?

**Comment by danielwrobert** [comment]
@scruffian The Woo and Jetpack blocks do appear in the Global Styles interface. That is expected, at least as far as I understand. The issue is, when applying those Global Styles to one of the Jetpack or Woo blocks, the styles are not saved to the frontend (see screenshots below). In my testing, this issue is present with the Gutenberg feature plugin enabled but not an issue with Core and no Gutenberg plugin active.

| Global Styles | With GB Plugin | W/O GB Plugin |
| ------ | ----- | ----- |
|  ![CleanShot 2022-09-08 at 11 27 04@2x](https://user-images.githubusercontent.com/481776/189163245-d447f3a9-3d74-4e24-a65c-748e22e73ee9.png)  |  ![CleanShot 2022-09-08 at 11 27 25@2x](https://user-images.githubusercontent.com/481776/189163372-6712cebb-cdde-4193-9001-bf23386eeda3.png)  |  ![CleanShot 2022-09-08 at 11 27 40@2x](https://user-images.githubusercontent.com/481776/189163517-2348b001-5385-4fb6-ad54-2213a64dc955.png)  |

**Comment by scruffian** [comment]
I created a fix for this here: https://github.com/WordPress/gutenberg/pull/44018

**Comment by ockham** [comment]
Closing. This was fixed in Gutenberg 14.1 by #44018, and for WordPress 6.1 by https://github.com/WordPress/wordpress-develop/pull/3529.
